ci-operator/config/openshift-eng/rebasebot/openshift-eng-rebasebot-main.yaml

```diff
@@ -59,6 +59,71 @@ tests:
         requests:
           cpu: 100m
           memory: 200Mi
+- as: cluster-api-provider-azure
+  cron: 0 12 * * Mon,Thu
+  steps:
+    test:
+    - as: cluster-api-provider-azure
+      commands: |
+        cat <<'EOF' > generate-manifests.sh
+        #!/bin/bash
+        set -e  # Exit immediately if a command exits with a non-zero status
+        set -o pipefail  # Return the exit status of the last command in the pipe that failed
+
+        stage_and_commit(){
+            # If commiter email and name is passed as environment variable then use it.
+            if [[ -z "$REBASEBOT_GIT_USERNAME" || -z "$REBASEBOT_GIT_EMAIL" ]]; then
+                author_flag=()
+            else
+                author_flag=(--author="$REBASEBOT_GIT_USERNAME <$REBASEBOT_GIT_EMAIL>")
+            fi
+
+            if [[ -n $(git status --porcelain) ]]; then
+                git add -A
+                git commit "${author_flag[@]}" -q -m "UPSTREAM: <drop>: Generate OpenShift manifests"
+            fi
+        }
+
+        # $REBASEBOT_SOURCE is version tag in format v0.0.0
+        if [[ -z "$REBASEBOT_SOURCE" ]]; then
+            echo "The environment variable REBASEBOT_SOURCE is not set." >&2
+            exit 1
+        fi
+
+        # Remove the leading 'v' from REBASEBOT_SOURCE
+        export PROVIDER_VERSION="${REBASEBOT_SOURCE#v}"
+
+        pushd openshift
+        make ocp-manifests
+        popd
+        stage_and_commit
+        EOF
+        chmod +x generate-manifests.sh
```

Comment:
This is fine for now, but it'd be nice to have it as a builtin lifecyclehook as it is common across all the capi providers

Reply:
We already agreed on the design and implementation for built in hooks, they won't be part of the rebasebot code, but a separate place: https://github.com/openshift-eng/rebasebot?tab=readme-ov-file#builtin-lifecycle-hook-scripts

Other project specific hooks can be stored in their own repos like so: https://github.com/openshift-eng/rebasebot?tab=readme-ov-file#scripts-stored-inside-the-repository

Reply:
Ok, I think built in scripts can make sense for general use cases, but for CAPI specific stuff that we need to do, does that really make sense?

Reply:
Yes, re-looking now at this code, it is CAPI generator specific, so I think having this script somewhere in the cluster-capi-operator is more sensible.

Another story is the `builtin-get-latest-stable-version.sh` from below which is generic and can be used across many different usecases.